Linux Docker build failed since v4.8.3

### Steps to reproduce

1. Do clone repository
`git clone https://github.com/telegramdesktop/tdesktop.git
cd tdesktop
git checkout v4.8.3 # dev, v4.11.7, v4.9.9
git submodule update --init --recursive
`
2. Build docker image
`./Telegram/build/prepare/linux.sh`
3. Run build
`docker run --rm -it \
        -v $PWD:/usr/src/tdesktop \
        tdesktop:centos_env \
        /usr/src/tdesktop/Telegram/build/docker/centos_env/build.sh \
        -D TDESKTOP_API_ID=YOUR_API_ID \
        -D TDESKTOP_API_HASH=YOUR_API_HASH`

dev - error
v4.11.7 - error
v4.8.3 - error
v4.9.9 -  error

v4.7.1 - ok, but
./Telegram
./Telegram: symbol lookup error: ./Telegram: undefined symbol: g_once_init_leave_pointer

v4.1.2 - ok

Compilation output:

[dev.txt](https://github.com/telegramdesktop/tdesktop/files/13515394/dev.txt)
[v4.8.3.txt](https://github.com/telegramdesktop/tdesktop/files/13515395/v4.8.3.txt)


### Expected behaviour

successful build

### Actual behaviour

Compilation error:
`
FAILED: Telegram/lib_base/CMakeFiles/lib_base.dir/Release/base/platform/linux/base_system_media_controls_linux.cpp.o
...
In file included from /usr/src/tdesktop/out/gen/glib/glib.hpp:48,
                 from /usr/src/tdesktop/out/gen/gobject/gobject.hpp:14,
                 from /usr/src/tdesktop/out/gen/gio/gio.hpp:14,
                 from /usr/src/tdesktop/out/gen/mpris/mpris.hpp:14,
                 from /usr/src/tdesktop/Telegram/lib_base/base/platform/linux/base_system_media_controls_linux.cpp:12:
/usr/src/tdesktop/out/gen/glib/_enums.hpp:1759:47: error: ‘GWin32OSType’ was not declared in this scope
 1759 | enum class Win32OSType : std::underlying_type<GWin32OSType>::type {
      |                                               ^~~~~~~~~~~~
/usr/src/tdesktop/out/gen/glib/_enums.hpp:1759:59: error: template argument 1 is invalid
 1759 | enum class Win32OSType : std::underlying_type<GWin32OSType>::type {
      |                                                           ^
/usr/src/tdesktop/out/gen/glib/_enums.hpp:1759:62: error: expected ‘;’ or ‘{’ before ‘type’
 1759 | enum class Win32OSType : std::underlying_type<GWin32OSType>::type {
      |                                                              ^~~~
/usr/src/tdesktop/out/gen/glib/_enums.hpp:1760:3: error: ‘ANY_’ was not declared in this scope
...
`

### Operating system

debian-12, debian-11

### Version of Telegram Desktop

all versions above v4.8.3

### Installation source

Static binary from official website

### Crash ID

_No response_

### Logs

_No response_

--- a/Telegram/build/docker/centos_env/Dockerfile
+++ b/Telegram/build/docker/centos_env/Dockerfile
@@ -703,7 +703,22 @@ RUN git clone -b xkbcommon-1.6.0 --depth=1 {{ GIT }}/xkbcommon/libxkbcommon.git 
 	&& cd .. \
 	&& rm -rf libxkbcommon
 
-FROM patches AS glibmm
+FROM patches AS glib
+RUN git clone -b 2.78.1 --depth=1 {{ GIT }}/GNOME/glib.git \
+	&& cd glib \
+	&& meson build \
+		--buildtype=plain \
+		--default-library=both \
+		-Dtests=false \
+		-Dmm-common:use-network=true \
+	&& meson compile -C build \
+	&& DESTDIR="{{ LibrariesPath }}/glib-cache" meson install -C build \
+	&& cd .. \
+	&& rm -rf glib
+
+FROM builder AS glibmm
+COPY --link --from=glib {{ LibrariesPath }}/glib-cache /
+
 RUN git clone -b 2.78.0 --depth=1 {{ GIT }}/GNOME/glibmm.git \
 	&& cd glibmm \
 	&& git apply ../patches/glibmm.patch \
@@ -715,7 +730,6 @@ RUN git clone -b 2.78.0 --depth=1 {{ GIT }}/GNOME/glibmm.git \
 		-Dsigc++-3.0:build-documentation=false \
 		-Dsigc++-3.0:build-examples=false \
 		-Dsigc++-3.0:validation=false \
-		-Dglib:tests=false \
 		-Dmm-common:use-network=true \
 	&& meson compile -C build \
 	&& DESTDIR="{{ LibrariesPath }}/glibmm-cache" meson install -C build \
@@ -723,7 +737,7 @@ RUN git clone -b 2.78.0 --depth=1 {{ GIT }}/GNOME/glibmm.git \
 	&& rm -rf glibmm
 
 FROM builder AS gobject-introspection
-COPY --link --from=glibmm {{ LibrariesPath }}/glibmm-cache /
+COPY --link --from=glib {{ LibrariesPath }}/glib-cache /
 
 RUN git clone -b 1.78.1 --depth=1 {{ GIT }}/GNOME/gobject-introspection.git \
 	&& cd gobject-introspection \
@@ -880,6 +894,7 @@ COPY --link --from=ffmpeg {{ LibrariesPath }}/ffmpeg-cache /
 COPY --link --from=openal {{ LibrariesPath }}/openal-cache /
 COPY --link --from=openssl {{ LibrariesPath }}/openssl-cache /
 COPY --link --from=xkbcommon {{ LibrariesPath }}/xkbcommon-cache /
+COPY --link --from=glib {{ LibrariesPath }}/glib-cache /
 COPY --link --from=glibmm {{ LibrariesPath }}/glibmm-cache /
 COPY --link --from=gobject-introspection {{ LibrariesPath }}/gobject-introspection-cache /
 COPY --link --from=qt {{ LibrariesPath }}/qt-cache /

--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -259,6 +259,27 @@ parts:
     after:
       - mozjpeg
 
+  glib:
+    source: https://github.com/GNOME/glib.git
+    source-depth: 1
+    source-tag: 2.78.1
+    plugin: meson
+    build-packages:
+      - mm-common
+      - ninja-build
+    meson-parameters:
+      - --buildtype=release
+      - --prefix=/usr
+      - --strip
+      - -Dtests=false
+      - -Dintrospection=disabled
+    prime:
+      - -./usr/include
+      - -./usr/lib/$CRAFT_ARCH_TRIPLET/pkgconfig
+      - -./usr/lib/$CRAFT_ARCH_TRIPLET/*.so
+    after:
+      - meson-deps
+
   glibmm:
     source: https://github.com/GNOME/glibmm.git
     source-depth: 1
@@ -277,14 +298,13 @@ parts:
       - -Dsigc++-3.0:build-documentation=false
       - -Dsigc++-3.0:build-examples=false
       - -Dsigc++-3.0:validation=false
-      - -Dglib:tests=false
-      - -Dglib:introspection=disabled
     prime:
       - -./usr/include
       - -./usr/lib/$CRAFT_ARCH_TRIPLET/pkgconfig
       - -./usr/lib/$CRAFT_ARCH_TRIPLET/*.so
     after:
       - meson-deps
+      - glib
 
   libjxl:
     source: https://github.com/libjxl/libjxl.git
@@ -523,7 +543,7 @@ parts:
       - -./usr/mkspecs
       - -./usr/modules
     after:
-      - glibmm
+      - glib
       - mozjpeg
       - patches
 
@@ -608,5 +628,5 @@ parts:
       - -./usr/lib/$CRAFT_ARCH_TRIPLET/*.a
     after:
       - ffmpeg
-      - glibmm
+      - glib
       - mozjpeg

